recipe/meta.yaml

```diff
@@ -17,9 +17,13 @@ requirements:
     - {{ compiler('c') }}
     - pkg-config
     - cargo-bundle-licenses
+    - sysroot_linux-64 2.17  # [linux64]
 
   host:
     - libgit2
+    # to build against libgit2 we also need the pkgconfig files
+    # for zlib
+    - zlib
```

Comment:
This should be done upstream.

Reply:
`libgit2` references `zlib` in its pkgconfig file as a private requirement.

This can make sense if public headers of `libgit2` import `zlib`'s header because then `pkg-config --cflags libgit2` would need to output `-I<path-to-libgit2-headers> -I<path-to-zlib-headers> ...` to make things using `libgit2` headers compile properly (well, in our case it probably would work without `-I<path-to-zlib-headers>` when we have them all directly in `${PREFIX}/include` -- but in the general case you'd need that additional include path).

The path to the `zlib` headers (and potentially additional flags) would be given by `pkg-config --cflags` which needs `zlib.pc`.
That file is in `conda-forge::zlib` such that `conda-forge::libgit2` needs to depend on that package (if we had a `libgit2-devel`/`liblibgit2`? packages, i.e., `libgit2` runtime and compile/link libraries separate, then this would be clearer).

I'll open a PR for `libgit2` to add the `zlib` dependency.

Reply:
> This can make sense if public headers of `libgit2` import `zlib`'s header because then `pkg-config --cflags libgit2` would need to output `-I<path-to-libgit2-headers> -I<path-to-zlib-headers> ...` to make things using `libgit2` headers compile properly (well, in our case it probably would work without `-I<path-to-zlib-headers>` when we have them all directly in `${PREFIX}/include` -- but in the general case you'd need that additional include path).

So, the "if public headers of `libgit2` import `zlib`'s header" doesn't seem to hold true, see https://github.com/conda-forge/libgit2-feedstock/pull/73#issuecomment-1962382821 .